pkg/api/customization/clustertemplate/cluster_template.go

```diff
@@ -0,0 +1,124 @@
+package clustertemplate
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+	"time"
+
+	"github.com/rancher/norman/api/access"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/norman/types/convert"
+	"github.com/rancher/norman/types/values"
+	"github.com/rancher/rancher/pkg/namespace"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	managementschema "github.com/rancher/types/apis/management.cattle.io/v3/schema"
+	"github.com/rancher/types/client/management/v3"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Wrapper struct {
+	ClusterTemplates              v3.ClusterTemplateInterface
+	ClusterTemplateLister         v3.ClusterTemplateLister
+	ClusterTemplateRevisionLister v3.ClusterTemplateRevisionLister
+	ClusterTemplateRevisions      v3.ClusterTemplateRevisionInterface
+}
+
+func (w Wrapper) Formatter(apiContext *types.APIContext, resource *types.RawResource) {
+	resource.AddAction(apiContext, "disable")
+	resource.Links["revisions"] = apiContext.URLBuilder.Link("revisions", resource)
+}
+
+func (w Wrapper) ActionHandler(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	var template client.ClusterTemplate
+	if err := access.ByID(apiContext, &managementschema.Version, client.ClusterTemplateType, apiContext.ID, &template); err != nil {
+		return err
+	}
+	switch actionName {
+	case "disable":
+		obj, err := w.ClusterTemplates.GetNamespaced(namespace.GlobalNamespace, template.Name, v1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if obj.Spec.Disabled {
+			return nil
+		}
+		toUpdate := obj.DeepCopy()
+		toUpdate.Spec.Disabled = true
+
+		_, err = w.ClusterTemplates.Update(toUpdate)
+		if err != nil {
+			return err
```

Comment:
Do we usually return a bare error in handlers, or wrap it up in apiContext.WriteResponse?